cmd/undo-main.go

```diff
@@ -218,17 +223,23 @@ func undoURL(ctx context.Context, aliasedURL string, last int, recursive, dryRun
 
 		if lastObjectPath != content.URL.Path {
 			// Print any object in the current list before reinitializing it
-			exitErr = undoLastNOperations(ctx, clnt, perObjectVersions, last, dryRun)
+			if remove {
+				exitErr = undoLastNOperations(ctx, clnt, perObjectVersions, last, dryRun)
+			}
+			remove = true
 			lastObjectPath = content.URL.Path
 			perObjectVersions = []*ClientContent{}
 		}
-
+		if content.IsLatest && action == "DELETE" && !content.IsDeleteMarker || content.IsLatest && action == "PUT" && content.IsDeleteMarker {
```

Comment:
Do you want to club conditions as `((content.IsLatest && action == "DELETE" && !content.IsDeleteMarker) || (content.IsLatest && action == "PUT" && content.IsDeleteMarker))` for better readability?